File: src/_modules/user/user.model.js
import mongoose from "mongoose";

const UserSchema = new mongoose.Schema(
    {
        name: {
            type: String,
            required: true
        },
        email: {
            type: String,
            required: true,
            unique: true, // Garante que o email seja único no banco de dados
            immutable: true // Garante que o email não possa ser alterado após a criação do usuário
        },
        password: {
            type: String,
            required: true
        },
        premium: {
            type: Boolean, 
            required: true,
            default: false // Define o padrão do usuário como não premium
        }
    }
);

const User = mongoose.model("User", UserSchema);

export default User;

File: wallettracker_mobile/lib/models/user_model.dart
class UserModel {
  final String id;
  final String name;
  final String email;
  final bool premium;

  UserModel({
    required this.id,
    required this.name,
    required this.email,
    required this.premium,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      id: json['id'],
      name: json['name'],
      email: json['email'],
      premium: json['premium'],
    );
  }
}